presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ZOrderIntegers
+{
+    private ZOrderIntegers() {}
+
+    // TODO: make int and long z-address version
+    public static byte[] encode(List<Integer> input, List<Integer> bitRanges)
+            throws Exception
+    {
+        if (input.size() > 10) {
+            throw new Exception("Z-ordering does not support more than 10 input numbers");
+        }
+
+        // Transfer into array for faster access and to change values
+        int[] inputList = new int[input.size()];
+        for (int index = 0; index < input.size(); index++) {
+            inputList[index] = input.get(index);
+            if (inputList[index] < 0) {
+                throw new Exception("Negative numbers are not supported for z-ordering");
+            }
+        }
```

Comment:
1. Transfer into array is not necessary; though the type of "input" is a List, but its implementation can be ArrayList.
2. While, this part of the code does two things: a. transfer; b check negative; as makes the code logic complex. 

So, after we remove the transfer logic, check negativity will naturally become a single task. Meanwhile, please consider wrap the logic into a function.